### DISCUSSION TITLE
Best way to provide near real-time audio stream to website

### DISCUSSION THREAD

**Discussion by GitchNZ** [rtl-airband/RTLSDR-Airband#473] [Q&A]
I've been working on this now for several weeks, and between my poor coding abilities and ChatGPT's limited suggestions, I'm not really getting very far. What I want to do is provide a website where I can listen to the output from rtl_airband in real-time wherever I am. I am currently doing this with OpenWebRX, but that doesn't support scan mode, so you're limited in what you can do with it.

Here's what I've tried so far with rtl_airband:

Shoutcast. This worked fine, but the lag was unacceptably high (around 5 seconds at the absolute least, usually more)

PulseAudio to VLC. This initially struggled with the stream stopping during periods of silence, so I used play to generate some very quiet white noise in the background, then pointed VLC at the loopback sound interface in pulseaudio. This worked, but only when I hand-fed all the commands. Between VLC not wanting to be run as root, and PulseAudio expecting to be run in user mode, I was never able to get all the links to start as a system daemon, so gave up.

I'm thinking the UDP stream mode must have been added for a reason, but I can't find any config online from anyone who's been able to make use of it, other than the one config that used ffplay to play the stream locally. 

What I want is basically what OpenWebRX provides, as far as audio is concerned - a web page that plays a real-time stream of the output from rtl_airband. I'm not at all worried about providing control for the radio. I just want the output to be playable as close to real-time as possible.

**Comment by R1ch1e-Lee** [comment]
I had Icecast installed locally on the same machine as rtlsdr-airband and I don't remember much delay or lag when listening remotely from that same machine.
